src/components/engagement/engagement.service.ts

```diff
@@ -984,6 +977,13 @@ export class EngagementService {
       }
       throw new ServerException('Could not create Internship Engagement');
     }
+    // InternshipEngagements can only be created on InternshipProjects
+    const projectType = await this.getProjectTypeById(projectId);
+
+    if (projectType && projectType !== ProjectType.Internship) {
+      throw new BadRequestException('That Project type is not Intership');
+    }
```

Comment:
Why'd you move this? It was fine where it was before. Currently this will give an error back even though the db has been changed.

Reply:
Oh, yes. You are right. There was another confusion lol. I changed again and fixed test case.